contracts/staking/src/tokenfactory/miniwasm.rs

```diff
@@ -0,0 +1,57 @@
+use cosmwasm_std::{Binary, Coin, CosmosMsg, StdError};
+use initia_proto::miniwasm::tokenfactory::v1::{MsgBurn, MsgCreateDenom, MsgMint};
+use initia_proto::traits::MessageExt;
+
+pub fn create_denom(sender: String, subdenom: String) -> Result<CosmosMsg, StdError> {
+    let bytes = MsgCreateDenom { sender, subdenom }
+        .to_bytes()
+        .map_err(|_| StdError::generic_err("Failed to serialize MsgCreateDenom"))?;
+    Ok(CosmosMsg::Stargate {
```

Comment:
Is it possible to return `MsgCreateDenom` type directly like Osmosis'?

Reply:
Do you mean having this return `MsgCreateDenom` directly? If so, we can do that, but it would result in functions returning different types depending on whether the contract is compiled with the `miniwasm` feature. With the current approach, however, the actual message is abstracted into a `CosmosMsg`, ensuring that all functions in `tokenfactory/osmosis.rs` and `tokenfactory/miniwasm.rs` have the same return type. 

I believe this approach is better as it helps prevent compilation issues when enabling different features and makes it easier to support other chains that have forked the `x/tokenfactory` module.

Reply:
I mean can we avoid using `CosmosMsg::Stargate`, just like how we're doing it for Osmosis:

```rust
pub fn mint(sender: String, amount: Coin, mint_to_address: String) -> Result<CosmosMsg, StdError> {
    Ok(MsgMint {
        sender,
        amount: Some(osmosis_std::types::cosmos::base::v1beta1::Coin {
            denom: amount.denom,
            amount: amount.amount.to_string(),
        }),
        mint_to_address,
    }
    .into())
}
```

I just saw there's `.into()` at the end, so miniwasm's `MsgCreateDenom` should support that. If not, I'm ok with the current impl.

Reply:
Unfortunately, the messages provided by the `initia-proto` package do not include the conversion to a `CosmosMsg`, so manual creation of a `CosmosMsg::Stargate` is necessary.

Reply:
Understood. Thanks!